docs/developers/architecture/app_model.md

```diff
@@ -153,9 +153,13 @@ Inline (lambda) and nested functions should be avoided for command callbacks to
 prevent memory leakage.
 ```
 
-All commands have a string id (e.g. '`napari:layer:duplicate`'). These are
-currently `CommandID` enums but we are considering changing to using
-plain strings to simplify decalaration of `Action`s.
+All commands have a string id (e.g. '`napari.window.file.open_files_dialog`').
+Some naming conventions:
+
+* use `.` to separate words
+* start with `napari` for builtin commands or the plugin name for plugin commands
+* `napari.window` prefix indicates the `napari` command is part of a GUI menu
```

Comment:
Is this only for `napari` commands? We don't have this convention for plugins, or should we?

Reply:
Also I noticed that our layer actions do not have `window` but they are part of a menu

Reply:
```suggestion
* the `napari.window` prefix indicates the `napari` command is part of a GUI menu
```

Reply:
I don't think anyone has thought about it deeply before (?), so we can discuss. Maybe it's meant to indicate a *menubar* menu item?

I'm also not sure why command ids aren't just `package:fully.qualified.python.name`?

Reply:
Honestly I am just following what was left by Talley.

Sometimes we use the same command, e.g., a lot of help menu items just open a URL. I guess in that case we can append an extra word.

I am not opposed to fully qualified python name:

* It would probably be equally long as current notion - (which is fine, shorter would be nice just style wise)
* It would follow plugin naming convention
* No extra info about where this command is (i.e., which menu)

Reply:
I don't agree with fully qualifying the python name. If we move forward with allowing actions to take kwargs, then multiple actions might share the same name. Similarly we sometimes wrap actions in napari stuff, and we may not have a fully qualified python name. I much prefer having the command ID give me a sense of where the command lives in the viewer, rather than in the code.

Reply:
I will say I don't love `napari.window` either. I think it should be `napari.menus.menu` for stuff that lives in a menu (including a context menu I guess?) and then just `napari.callback_name` for stuff that doesn't live in menus. I think everything does now maybe but longer term

Reply:
I guess the only pro for python name is that it would follow what is happening in plugin manifests, though maybe we want to update that? With NAP-6, maybe we want the command ID to indicate where in the menu the action will live?